Sources/Jetworking/ClientConfiguration.swift

```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+public struct ClientConfiguration {
+    let baseURL: URL
+    let authenticationMethod: AuthenticationMethod
+    let encoder: JSONEncoder
```

Comment:
In the future we should consider if we just use Encoder and Decoder. The Api could return xml for example.

Reply:
Unfortunately this is not possible as `JSONEncoder` or `JSONDecoder` are not dependent on `Encoder` or `Decoder`. Furthermore when using `Codable` objects are not usable with xml responses so this has to be dealt with in a special way. For now I would like to stick with the most common use case of having JSON data to send and retrieve.

Reply:
Yep that's totally fine!